src/queryResult/queryResultWebViewController.ts

```diff
@@ -87,4 +187,47 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
     ): void {
         this._sqlOutputContentProvider = provider;
     }
+
+    public setExecutionPlanService(service: ExecutionPlanService): void {
+        this.executionPlanService = service;
+    }
+
+    public setUntitledDocumentService(
+        service: UntitledSqlDocumentService,
+    ): void {
+        this.untitledSqlDocumentService = service;
+    }
+
+    private async createExecutionPlanGraphs() {
+        if (!this.state.executionPlan) {
+            const planFile: ep.ExecutionPlanGraphInfo = {
+                graphFileContent: this._executionPlanContents,
+                graphFileType: ".sqlplan",
+            };
+            try {
+                this.state.executionPlan =
+                    await this.executionPlanService.getExecutionPlan(planFile);
+                this.state.executionPlanGraphs =
+                    this.state.executionPlan.graphs;
+                this.state.loadState = ApiStatus.Loaded;
+                this.state.totalCost = this.calculateTotalCost();
+            } catch (e) {
+                this.state.loadState = ApiStatus.Error;
+                this.state.errorMessage = e.toString();
+            }
+        }
+        this.updateState();
```

Comment:
Does state still need to be updated if `this.state.executionPlan === undefined`?  Should `updateState()` be inside the if block?